### PR TITLE
fix toc links, update style guide for same, add "read-more" to "see-also" shortcode

### PR DESCRIPTION
**Pull request by mrjones-plip**
We had links in the TOC that didn't jump down the page, but instead jumped to another page or another site.  Fixed the instances of this as well as updated the style guide to explicitly guide future editors.

### CODE REVIEW

**Comment by mrjones-plip** [comment]
@n-orlowski  - since I'm making an update to the style guide, adding you as the reviewer, but am still feeling new to all this, so feedback welcome!

**Comment by mrjones-plip** [comment]
@abbyad  - FYI

**Comment by mrjones-plip** [comment]
> Cool! Thinking we might want to be consistent in using "read more" vs "see more" and the placement of these links? Ex. as a line break at the end of the content, "Read more about [linked item]"?

Yeah - I wasn't sure which way to go with this ("all the same" or "let's offer some variety!").  Should I make both the the [home page](https://github.com/medic/cht-docs/pull/203/files#diff-d7d320b626dac31f7a02ccd44e4f22aa) and the [architecture pages](https://github.com/medic/cht-docs/pull/203/files#diff-db041e36e93216f7408997bb40aa8ebc) I changed follow this style or just the home page? The architecture page seems lower profile given it's not, well, you know, the home page ;)

**Comment by abbyad** [comment]
We could make these more consistent by using something like the simple "see-also" shortcode that we created. For an example, check out the [Building Workflows page](https://docs.communityhealthtoolkit.org/apps/concepts/workflows/#sms-messaging):
image

We could use the See Also as is, or create another shortcode which is simply `read-more`, and is the "Read more" text on a new line with the link (and no page title).

**Comment by n-orlowski** [comment]
I think a `read-more` shortcode feels more appropriate for the homepage as it offers more detailed info vs `see-also` which provides relevant content 🙂

**Comment by mrjones-plip** [comment]
We have a preferred order for our [template locations](https://gohugo.io/templates/shortcode-templates/#shortcode-template-lookup-order)?  i.e. should we use `/layouts/shortcodes/<SHORTCODE>.html` or `/themes/<THEME>/layouts/shortcodes/<SHORTCODE>.html` for our new pancy fants `read-more` shortcode? I'm deferring to the former for now, but trivial to move to the latter!

**Comment by n-orlowski** [comment]
Would refer to @abbyad for that one!

**Comment by mrjones-plip** [comment]
OK, I have the new shortcode working on this branch.  tl;dr, just like `see-also`, only with 100% more `read-more` ;)

read more in action

I'll update the style guide next!

**Comment by mrjones-plip** [comment]
k - back to you @n-orlowski!  I'm not sure how I qualify the difference [between the two shortcodes](https://github.com/medic/cht-docs/pull/203/files#diff-7db4f3c09183c397906b8d3248a43bcaR156), so feedback welcome!

**Comment by n-orlowski** [comment]
Looks good to me!

**Comment by mrjones-plip** [comment]
@abbyad  - good to go? I know we didn't add you as a reviewer, but wanted to check in real quick ;)

**Comment by mrjones-plip** [comment]
> We have a preferred order for our [template locations](https://gohugo.io/templates/shortcode-templates/#shortcode-template-lookup-order)? i.e. should we use `/layouts/shortcodes/<SHORTCODE>.html` or `/themes/<THEME>/layouts/shortcodes/<SHORTCODE>.html` for our new pancy fants `read-more` shortcode? I'm deferring to the former for now, but trivial to move to the latter!

So I shot myself in the foot earlier when I accidentally nuked `/themes/<THEME>` and then remembered that we [include sub resources](https://github.com/medic/cht-docs#building-the-documentation) (eg `git submodule update --init --recursive`) so this isn't the place to put templates like this new one.  We should always us `/layouts/shortcodes/`. I guess FYI @n-orlowski b/c Marc already knew ;)

File: content/en/contribute/docs/style-guide.md [hunk 1]
```diff
@@ -144,13 +144,18 @@ Connecting readers to related content in different pages is an important aspect
...
+   
+1. **See Also**: the `see-also` shortcode is available to connect to an important concept within the documentation site. The link will be more prominent to the reader by having a common prefix and shown on a separate line.
    
    For example, `{{</* see-also page="design/icons" */>}}` will show as seen here: {{< see-also page="design/icons" >}}
+   
+   You can also make the callout say "Read More" with the `prefix` tag: `{{</* see-also prefix="Read More" page="design/icons" */>}}`. This will show as seen here: {{< see-also page="design/icons" prefix="Read More" >}} 
+   
+   A custom title and anchor can be provided with this as well. For example, `{{</* see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" */>}}`, will show as: {{< see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" >}}
+   
+   Please use `see-also` when reference __related topics__ (e.g. [Workflows]({{< ref "apps/concepts/workflows" >}})) and use `read-more` when referencing the __same topic__ in more depth  (e.g. [Home Page]({{< ref "/" >}})) .
```
Comment (abbyad) [inline content/en/contribute/docs/style-guide.md:156]: We don't use `__text__` for bold (much?) anywhere else in documentation, and don't have it listed as such in the [docs style guide](https://github.com/medic/cht-docs/blob/74d07696a3c41de337b1673e850663862e5ef32a/content/en/contribute/docs/style-guide.md#use-markdown-notation). It is a bit arbitrary, but unless there is a clear reason to use double underscores it would be nice to stay consistent with the styling of the page and use `**text**` here. 

We can also add this as a do or don't in the style guide if you think that would help!

Comment (mrjones-plip) [reply content/en/contribute/docs/style-guide.md]: Oop! I'd actually meant them to be italics, not bold. Moved back to italics, but am happy to go bold (via `**`) if that's preferred!

**Comment by mrjones-plip** [comment]
@abbyad - i've settled on a working shortcode template (hopefully).  Lemme know what you think! I tried to strike a balance between legible and hugo not injecting random `<p>` tags in the output - very open to feedback!

File: layouts/shortcodes/see-also.html [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+    {{ else }}
+        {{ errorf "[see-also] REF_NOT_FOUND: Ref \"%s\": \"%s\": page not found" $page $save.Page.File.Path }}
+    {{ end }}
 {{ end }}
+
+<p>
+    <em>
+    {{ ( or (.Get "prefix") "See Also" ) }}:
+    </em>{{ with $.Site.GetPage $page }}
+    {{ $finaltile := .Title }}{{ $finalhref := (ref . $page) }}{{ $finalanchor := "null" }}{{ if $title }}{{ $finaltile = $title }}{{ end }}{{ if $anchor }}<a href="{{ $finalhref }}#{{ $anchor }}" >{{ $finaltile }}</a>
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply layouts/shortcodes/see-also.html]: yes, thank you!

File: layouts/shortcodes/see-also.html [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
+        {{ warnf "[see-also] REF_NOT_FOUND: Ref \"%s\": \"%s\": page not found" $page $save.Page.File.Path }}
+    {{ else }}
+        {{ errorf "[see-also] REF_NOT_FOUND: Ref \"%s\": \"%s\": page not found" $page $save.Page.File.Path }}
+    {{ end }}
 {{ end }}
+
+<p>
+    <em>
+    {{ ( or (.Get "prefix") "See Also" ) }}:
+    </em>{{ with $.Site.GetPage $page }}
```
[the remark this answers is not in the document]
Comment (mrjones-plip) [reply layouts/shortcodes/see-also.html:17]: Agreed - However, I think I'm lacking a concept or three about how Hugo assembles these.  I'm _real_ good and getting compilation errors, not so good at writing easy to read templates.  I'll ping you to see if you're free for a walk through - I suspect 5 minutes of you time will add the mental glue I'm missing.

[the remark this answers is not in the document]
Comment (mrjones-plip) [reply layouts/shortcodes/see-also.html]: Whew - THANK YOU for that walk through this morning. I think we're at the "easy to read" level now and it all works with just one output of the link! Gonna resolve this, but do have one more question I'll add below.

File: layouts/shortcodes/see-also.html [hunk 4]
```diff
@@ -1,18 +1,19 @@
...
 {{ with $.Site.GetPage $page }}
-{{ $title := $save.Get "title" }}
-{{ if $title }}
-{{ printf "<p><em>See Also:</em> <a href=%s#%s>%s</a></p>" (ref . $page) $anchor $title | safeHTML }}
+    {{ $title := or ($save.Get "title") .Title }}
+    {{ $url := ref . $page }}
+    {{ with $anchor }}
+        {{ $url = print $url "#" $anchor }}
+    {{ end }}
+    <p><em>{{ $prefix }}</em>: <a href="{{ $url }}">{{ $title }}</a></p>
```
Comment (mrjones-plip) [inline layouts/shortcodes/see-also.html:12]: @abbyad - soooo... this indent looks pretty, and would be my preferred formatting, however if we use a `{{<` to including the shortcode like we do in the style guide, all good - everything works. However, in some places, and this tripped me up a whole lot in development, when we use `{{%` to include the shortcode, it actually renders the HTML:

inline code

I finally traced this back to the [0.55.0 release](https://github.com/gohugoio/hugo/releases/tag/v0.55.0) changing how they render short codes (I guess no more [blackfriday](https://gohugo.io/getting-started/configuration-markup)?).

tl;dr - we have 7 that use the `{{<` and 29 that use `{{%` - so I guess we should not indent so it works with both styles of code?

Comment (abbyad) [reply layouts/shortcodes/see-also.html:12]: Ah, that helps to clear this up. Can you think of a reason to use `{{%` for this shortcode? Seems as though we should use `{{<` over `{{%`, and update the style guide and existing files -- you said you like `sed`, right? :)

Comment (mrjones-plip) [reply layouts/shortcodes/see-also.html:12]: The only reason I can think to use `{{%` is that it's a documented and accepted syntax and "because it formats things better" doesn't seem like _that_ big of a reason to not use.  

However, when we have thousands of documents in this repo, and a lot more of them are templates....maybe formatting becomes more important?!  So, yeah, like you `sed` I `sed` (big fan of puns ;), let's go for it! I'll swap 'em out in this PR - stay tuned!

**Comment by mrjones-plip** [comment]
K, to ease QA/Code Review, here's an exhaustive list of all the uses of `%` with `see-also`:

* [./content/en/core/_index.md](http://localhost:1313/core/):{{% see-also page="apps" %}}
* [./content/en/apps/features/contacts/_index.md](http://localhost:1313/apps/features/contacts/):{{% see-also page="contact-page" anchor="contact-summary" title="Defining Contact Summary" %}}
* [./content/en/apps/features/contacts/_index.md](http://localhost:1313/apps/features/contacts/):{{% see-also page="contact-page" anchor="condition-cards" title="Defining Condition Cards" %}}
* [./content/en/apps/features/contacts/_index.md](http://localhost:1313/apps/features/contacts/):{{% see-also page="contact-page" anchor="care-guides" title="Defining Care Guides" %}}
* [./content/en/apps/features/integrations/dhis2.md](http://localhost:1313/apps/features/integrations/dhis2/):{{% see-also page="apps/features/targets" title="Targets" %}}
* [./content/en/apps/concepts/hierarchy.md](http://localhost:1313/apps/concepts/hierarchy/):{{% see-also page="apps/reference/app-settings/hierarchy" title="Defining Hierarchy" %}}
* [./content/en/apps/concepts/workflows.md](http://localhost:1313/apps/concepts/workflows/):{{% see-also page="apps/reference/tasks" title="Defining Tasks" %}}
* [./content/en/apps/concepts/workflows.md](http://localhost:1313/apps/concepts/workflows/):{{% see-also page="apps/reference/app-settings" anchor="sms-workflows" title="Defining SMS Workflows" %}}
* [./content/en/apps/concepts/workflows.md](http://localhost:1313/apps/concepts/workflows/):{{% see-also page="apps/reference/app-settings/outbound" title="Outbound Push" %}}
* [./content/en/apps/concepts/users.md](http://localhost:1313/apps/concepts/users/):{{% see-also page="apps/reference/app-settings/user-roles" anchor="" title="Defining User Roles" %}}
* [./content/en/apps/concepts/users.md](http://localhost:1313/apps/concepts/users/):{{% see-also page="apps/reference/app-settings/user-permissions" title="Defining User Permissions" %}}
* [./content/en/apps/concepts/access.md](http://localhost:1313/apps/concepts/access/):{{% see-also page="apps/concepts/navigation" %}}
* [./content/en/apps/guides/database/_index.md](http://localhost:1313/apps/guides/database/):{{% see-also page="core/overview/db-schema" %}}
* [./content/en/apps/guides/database/_index.md](http://localhost:1313/apps/guides/database/):- "telemetry". Aggregate telemetry information for performance and usage metrics analysis. {{% see-also page="apps/guides/performance/telemetry" %}}
* [./content/en/apps/guides/database/_index.md](http://localhost:1313/apps/guides/database/):{{% see-also page="apps/guides/performance/replication" %}}
* [./content/en/apps/guides/database/_index.md](http://localhost:1313/apps/guides/database/):{{% see-also page="core/overview/data-flows-for-analytics" %}}
* [./content/en/apps/guides/forms/contact-forms.md](http://localhost:1313/apps/guides/forms/contact-forms/):{{% see-also page="apps/reference/forms/contact" title="Reference Documentation for Contact Forms" %}}
* [./content/en/apps/guides/integrations/dhis2-aggregate.md](http://localhost:1313/apps/guides/integrations/dhis2-aggregate/):{{% see-also page="apps/concepts/hierarchy" title="Hierarchies" %}}
* [./content/en/apps/guides/integrations/dhis2-aggregate.md](http://localhost:1313/apps/guides/integrations/dhis2-aggregate/):{{% see-also page="apps/concepts/forms" title="Forms" %}}
* [./content/en/apps/guides/integrations/dhis2-aggregate.md](http://localhost:1313/apps/guides/integrations/dhis2-aggregate/):{{% see-also page="apps/reference/app-settings/dhis2" title="DHIS2 in App Settings" %}}
* [./content/en/apps/guides/integrations/dhis2-aggregate.md](http://localhost:1313/apps/guides/integrations/dhis2-aggregate/):{{% see-also page="apps/reference/forms/contact" title="Contact Forms" %}}
* [./content/en/apps/guides/integrations/dhis2-aggregate.md](http://localhost:1313/apps/guides/integrations/dhis2-aggregate/):{{% see-also page="apps/reference/targets" title="Targets" %}}
* [./content/en/apps/guides/integrations/dhis2-aggregate.md](http://localhost:1313/apps/guides/integrations/dhis2-aggregate/):{{% see-also page="apps/concepts/users" title="Users" %}}
* [./content/en/apps/reference/app-settings/dhis2.md](http://localhost:1313/apps/reference/app-settings/dhis2/):{{% see-also page="apps/features/integrations/dhis2" title="DHIS2 Integration" %}}
* [./content/en/apps/reference/tasks.md](http://localhost:1313/apps/reference/tasks/):{{% see-also page="apps/features/tasks" title="Tasks Overview" %}}
* [./content/en/apps/reference/forms/app.md](http://localhost:1313/apps/reference/forms/app/):{{% see-also page="apps/reference/forms/contact" anchor="care-guides" title="Passing contact data to care guides" %}}
* [./content/en/apps/reference/forms/app.md](http://localhost:1313/apps/reference/forms/app/):{{% see-also page="design/apps" anchor="content-and-layout" title="Content and Layout" %}}
* [./content/en/apps/reference/targets.md](http://localhost:1313/apps/reference/targets/):{{% see-also page="apps/features/targets" title="Targets Overview" %}}
* [./content/en/apps/reference/resources/_index.md](http://localhost:1313/apps/reference/resources/):{{% see-also page="design/icons" title="Icon Library" %}}

**Comment by mrjones-plip** [comment]
@abbyad  - ready for you! `grep -r '{{% see-also' content/en` should show clean and all above links should not have janky inline `html` - just beautiful rendered `see-also` and `read-more`.

I have to come clean - I did end up using `sed`, so three nerd demerits for me.  I used the regex search and replace in my IDE:
> `\{\{% see-also ([a-zA-Z0-9- ="/]*) %}}`
> `{{< see-also $1 >}}`

File: content/en/contribute/docs/style-guide.md [hunk 5]
```diff
@@ -144,13 +144,18 @@ Connecting readers to related content in different pages is an important aspect
...
+   
+1. **See Also**: the `see-also` shortcode is available to connect to an important concept within the documentation site. The link will be more prominent to the reader by having a common prefix and shown on a separate line.
    
    For example, `{{</* see-also page="design/icons" */>}}` will show as seen here: {{< see-also page="design/icons" >}}
+   
+   You can also make the callout say "Read More" with the `prefix` tag: `{{</* see-also prefix="Read More" page="design/icons" */>}}`. This will show as seen here: {{< see-also page="design/icons" prefix="Read More" >}} 
+   
+   A custom title and anchor can be provided with this as well. For example, `{{</* see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" */>}}`, will show as: {{< see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" >}}
+   
+   Please use `see-also` when reference _related topics_ (e.g. [Workflows]({{< ref "apps/concepts/workflows" >}})) and use `read-more` when referencing the _same topic_ in more depth  (e.g. [Home Page]({{< ref "/" >}})) .
```
Comment (abbyad) [inline content/en/contribute/docs/style-guide.md:156]: Fixed minor typo, and [we should avoid use of latin](https://docs.communityhealthtoolkit.org/contribute/docs/style-guide/#latin-phrases). I've made a suggested edit, feel free to correct if you had intended it differently.

```suggestion
   Please use `see-also` when referencing _related topics_ , as seen in [Workflows]({{< ref "apps/concepts/workflows" >}}), and use `read-more` when referencing the _same topic_ in more depth, as in the [Home Page]({{< ref "/" >}}).  
```

File: content/en/contribute/docs/style-guide.md [hunk 6]
```diff
@@ -144,13 +144,18 @@ Connecting readers to related content in different pages is an important aspect
...
-2. **See Also**: the `see-also` shortcode is available to connect to an important concept within the documentation site. The link will be more prominent to the reader by having a common prefix and shown on a separate line.
+   
+1. **See Also**: the `see-also` shortcode is available to connect to an important concept within the documentation site. The link will be more prominent to the reader by having a common prefix and shown on a separate line.
    
    For example, `{{</* see-also page="design/icons" */>}}` will show as seen here: {{< see-also page="design/icons" >}}
+   
+   You can also make the callout say "Read More" with the `prefix` tag: `{{</* see-also prefix="Read More" page="design/icons" */>}}`. This will show as seen here: {{< see-also page="design/icons" prefix="Read More" >}} 
+   
+   A custom title and anchor can be provided with this as well. For example, `{{</* see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" */>}}`, will show as: {{< see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" >}}
```
Comment (abbyad) [inline content/en/contribute/docs/style-guide.md:154]:
```suggestion
   A custom title and anchor can be provided as well. For example, `{{</* see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" */>}}`, will show as: {{< see-also page="design/icons" title="Learn about the Icon Library" anchor="about-the-icon-library" >}}
```

**Comment by mrjones-plip** [comment]
Awesome!  Thanks so much for working through this PR with me - super helpful! I'll merge to master tomorrow AM - just so I'm around when it goes live.
